test/test_base_service.py

```diff
@@ -647,6 +647,34 @@ def test_gzip_compression():
     assert prepped['headers'].get('content-encoding') == 'gzip'
 
 
+def test_gzip_compression_file_input():
```

Comment:
Had to put the new test cases into a separate function to avoid the `too-many-branches` linter error.